tests/library/browsercontext-fetch.spec.ts

```diff
@@ -218,6 +218,33 @@ it('should add cookies from Set-Cookie header', async ({ context, page, server }
   expect((await page.evaluate(() => document.cookie)).split(';').map(s => s.trim()).sort()).toEqual(['foo=bar', 'session=value']);
 });
 
+it('should preserve cookie order from Set-Cookie header', async ({ context, page, server, browserName, isLinux }) => {
+  it.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/23390' });
+  it.fixme(browserName === 'webkit' && isLinux);
+  server.setRoute('/setcookie.html', (req, res) => {
+    res.setHeader('Set-Cookie', ['cookie.0=foo', 'cookie.1=bar']);
+    res.end();
+  });
+  await page.request.get(server.PREFIX + '/setcookie.html');
+  const cookies = await context.cookies();
+  expect(cookies.map(c => ({ name: c.name, value: c.value }))).toEqual([
```

Comment:
I don't think we should guarantee the order in `context.cookies()`.

Reply:
We don't provide such guarantees publicly, it's just an internal detail covered in this test.